packages/priority-queue/src/index.js

```diff
@@ -0,0 +1,50 @@
+const requestFrame = window.requestIdleCallback ? window.requestIdleCallback : window.requestAnimationFrame;
+
+export const createQueue = () => {
```

Comment:
It's interesting to observe how we in the JavaScript community resist classes when there's all the telltale signs of this being class-like (internal state, instance methods). There's certainly precedent (Redux's `createStore`), it's just interesting how we do this out of some motivation to be "functional" by avoiding `class`.

**Edit:** I guess one advantage here is in the availability to define `runWaitingList` as "private".